node_pools.tf

```diff
@@ -99,6 +99,98 @@ resource "google_container_node_pool" "node_pool_mt" {
   }
 }
 
+resource "google_container_node_pool" "node_pool_dynamic_pods" {
+
+  provider = google-beta
+
+  # these can't be created or deleted at the same time.
+  depends_on = [google_container_node_pool.node_pool_platform]
+  # version    = data.google_container_cluster.primary.master_version
+  version = var.kube_version_gke
+
+  # We want the dynamic-pods node pool to be completely replaced
+  # instead of rolling deployment.
+  # The master_version will ensure that the node pool is created then
+  # destroyed if there is an update.
+  name = "${var.deployment_id}-dp-${formatdate("MM-DD-hh-mm", timestamp())}"
```

Comment:
delete name, allow terraform to auto-name

Reply:
Done

Reply:
@sjmiller609 @kaxil what does omitting the names do for us? Makes it hard to see what node pools are for what.